src/plugins/template/src/infer_request.cpp

```diff
@@ -0,0 +1,253 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "infer_request.hpp"
+
+#include <algorithm>
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "openvino/core/except.hpp"
+#include "plugin.hpp"
+#include "template_itt.hpp"
+
+using Time = std::chrono::high_resolution_clock;
+
+namespace {
+
+static void AllocateImplSingle(std::vector<ov::Tensor>& tensors,
+                               size_t idx,
+                               const ov::element::Type& element_type,
+                               const ov::Shape& shape) {
+    OPENVINO_ASSERT(idx < tensors.size());
+    if (!tensors.at(idx) || tensors.at(idx).get_element_type() != element_type) {
+        tensors.at(idx) = ov::Tensor(element_type, shape);
+    } else {
+        tensors.at(idx).set_shape(shape);
+    }
+}
+
+static void AllocateImpl(std::vector<ov::Tensor>& tensors, const std::vector<ov::Output<const ov::Node>>& ports) {
+    OPENVINO_ASSERT(tensors.size() == ports.size());
+
+    for (size_t i = 0; i < tensors.size(); i++) {
+        AllocateImplSingle(tensors,
+                           i,
+                           ports.at(i).get_element_type(),
+                           ports.at(i).get_partial_shape().is_dynamic() ? ov::Shape{0} : ports.at(i).get_shape());
+    }
+}
+
+}  // namespace
+
+// ! [infer_request:ctor]
+TemplatePlugin::InferRequest::InferRequest(const std::shared_ptr<const TemplatePlugin::CompiledModel>& model)
+    : ov::ISyncInferRequest(model) {
+    // TODO: allocate infer request device and host buffers if needed, fill actual list of profiling tasks
+
+    auto compiled_model = std::const_pointer_cast<TemplatePlugin::CompiledModel>(get_template_model());
+    auto requestID = std::to_string(compiled_model->_requestId.fetch_add(1));
+
+    std::string name = compiled_model->m_model->get_friendly_name() + "_Req" + requestID;
+    m_profiling_task = {
+        openvino::itt::handle("Template" + std::to_string(compiled_model->_cfg.deviceId) + "_" + name + "_Preprocess"),
+        openvino::itt::handle("Template" + std::to_string(compiled_model->_cfg.deviceId) + "_" + name + "_Postprocess"),
+        openvino::itt::handle("Template" + std::to_string(compiled_model->_cfg.deviceId) + "_" + name +
+                              "_StartPipline"),
+        openvino::itt::handle("Template" + std::to_string(compiled_model->_cfg.deviceId) + "_" + name + "_WaitPipline"),
+    };
+
+    m_executable = compiled_model->get_template_plugin()->_backend->compile(compiled_model->m_model);
+
+    // Allocate plugin backend specific memory handles
+    m_plugin_input_tensors.resize(get_inputs().size());
+    m_plugin_output_tensors.resize(get_outputs().size());
+    m_input_tensors.resize(get_inputs().size());
+    m_output_tensors.resize(get_outputs().size());
```

Comment:
do we need two types of tensors here? 
I suppose with new API we don't need backend and IE ones